yew/src/agent/worker/queue.rs

```diff
@@ -0,0 +1,52 @@
+use std::cell::{RefCell, RefMut};
+use std::collections::{hash_map, HashMap, HashSet};
+use std::hash::Hash;
+
+/// Thread-local instance used to queue worker messages
+pub struct Queue<T: Eq + Hash> {
+    loaded_agents: RefCell<HashSet<T>>,
+    msg_queue: RefCell<HashMap<T, Vec<Vec<u8>>>>,
+}
+
+impl<T: Eq + Hash> Queue<T> {
+    pub fn new() -> Queue<T> {
+        Queue {
+            loaded_agents: RefCell::new(HashSet::new()),
+            msg_queue: RefCell::new(HashMap::new()),
+        }
+    }
+
+    #[inline]
+    pub fn borrow_msg_queue_mut(&self) -> RefMut<'_, HashMap<T, Vec<Vec<u8>>>> {
```

Comment:
I think a `drain_msg_queue` method would be nicer. It could even remove the `id` from the message queue map since the queue won't be needed anymore.

Reply:
For now I've implemented `remove_msg_queue` which returns the vector in question, creating an iterator would need a place to store the interator's ref, since we want to remove the msg_queue when getting the iterator

Reply:
Great, your implementation is what I was hoping for 👍